File: lib/domain/shared/dtos/reference_flag_change_denied_failure.rb
# frozen_string_literal: true

module Domain
  module Shared
    module Dtos
      # 参照フラグ変更がポリシーで拒否されたときに Output Port へ渡す。
      class ReferenceFlagChangeDeniedFailure
        attr_reader :message, :resource_id

        # @param message [String]
        # @param resource_id [Integer]
        def initialize(message:, resource_id:)
          @message = message
          @resource_id = resource_id
        end
      end
    end
  end
end

File: crates/agrr-domain/src/shared/dtos/reference_flag_change_denied_failure.rs
/// Ruby: `Domain::Shared::Dtos::ReferenceFlagChangeDeniedFailure`
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct ReferenceFlagChangeDeniedFailure {
    pub message: String,
    pub resource_id: i64,
}

impl ReferenceFlagChangeDeniedFailure {
    pub fn new(message: impl Into<String>, resource_id: i64) -> Self {
        Self {
            message: message.into(),
            resource_id,
        }
    }
}

#[cfg(test)]
mod dtos_reference_flag_change_denied_failure_test_inline {
    use super::*;
    include!(concat!(env!("CARGO_MANIFEST_DIR"), "/test/shared/dtos_reference_flag_change_denied_failure_test.rs"));
}
